app/src/main/java/com/aragh/kotlin2/screen/albumdetails/AlbumDetailsActivity.kt

```diff
@@ -0,0 +1,73 @@
+package com.aragh.kotlin2.screen.albumdetails
+
+import android.content.Context
+import android.content.Intent
+import android.os.Bundle
+import android.support.constraint.ConstraintSet
+import android.support.v7.app.AppCompatActivity
+import android.transition.TransitionManager
+import com.aragh.kotlin2.R
+import kotlinx.android.synthetic.main.activity_albumdetails.*
+import org.koin.android.ext.android.inject
+
+
+class AlbumDetailsActivity : AppCompatActivity(), ViewContract {
+
+  companion object {
+    private const val ALBUM_ID_EXTRA = "albumId"
+
+    fun intent(context: Context, albumId : Int) : Intent {
```

Comment:
You may consider adding `@JvmStatic` in case it's ever called from Java.. or f*k it xD

Reply:
i would rather not clutter it up here. I hope there is no danger of it ever being called from java